File: data/examples/python/bit_11.py
from inline import itest

def r_short(self) -> int:
    buf = self.r_string(2)
    x = buf[0]
    x |= buf[1] << 8
    itest().given(buf, [0, 1]).given(x, 0).check_eq(x, 256)
    x |= -(x & (1 << 15))  # Sign-extend
    itest().given(x, 256).check_eq(x, 256)

    return x


def r_long(self) -> int:
    buf = self.r_string(4)
    x = buf[0]
    x |= buf[1] << 8
    x |= buf[2] << 16
    x |= buf[3] << 24
    x |= -(x & (1 << 31))  # Sign-extend
    return x

File: data/examples/java/bit_11.java
public class Bit11 {
	public long getLong(byte[] b, int offset) {
		long v = b[offset + 7];
		for (int i = 6; i >= 0; i--) {
			v = (v << 8) | (b[offset + i] & 0xff);
            itest().given(b, "abcdefghijklmn".getBytes()).given(offset, 1).given(v, b[offset+7]).given(i, 6).checkEq(v, 26984L);
		}
		return v;
	}    
}
